lib/budgie_web/live/budget_show_live.html.heex

```diff
@@ -0,0 +1,84 @@
+<.modal
+  :if={@live_action in [:new_transaction, :edit_transaction]}
+  id="transaction-modal"
+  on_cancel={JS.navigate(~p"/budgets/#{@budget}", replace: true)}
+  show
+>
+  <.live_component
+    module={BudgieWeb.TransactionDialog}
+    id="transaction"
+    budget={@budget}
+    transaction={assigns[:transaction] || default_transaction()}
```

Comment:
A transaction is passed in for use in the validation and saving changesets.

When editing, this will be the transaction specified in the URL.
When creating, we fall back to a default transaction referencing the current UTC date.